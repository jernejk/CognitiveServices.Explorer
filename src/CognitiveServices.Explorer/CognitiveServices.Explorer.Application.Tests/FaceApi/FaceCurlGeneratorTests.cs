using CognitiveServices.Explorer.Application.FaceApi;
using FluentAssertions;
using Snapshooter.Xunit;
using Xunit;

namespace CognitiveServices.Explorer.Application.Tests.FaceApi
{
    public class FaceCurlGeneratorTests
    {
        [Fact]
        public void ShouldGenerateDetectBinary()
        {
            new FaceCurlGenerator()
                .Detect(new byte[1] { 5 })
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateDetectUrl()
        {
            new FaceCurlGenerator()
                .Detect("http://test.url")
                .Should()
                .MatchSnapshot();
        }

        [Fact]
        public void ShouldGenerateIdentify()
        {
            string[] faces = { "12234324", "34534253425" };
            new FaceCurlGenerator()
                .Identify("default", faces)
                .Should()
                .MatchSnapshot();
        }
    }
}
