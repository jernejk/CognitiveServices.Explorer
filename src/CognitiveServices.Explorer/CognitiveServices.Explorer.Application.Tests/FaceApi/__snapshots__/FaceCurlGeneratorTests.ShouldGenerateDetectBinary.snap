{
  "RequestName": null,
  "Headers": {},
  "RelativePath": "face/v1.0/detect",
  "Queries": {
    "recognitionModel": "recognition_01",
    "detectionModel": "detection_01"
  },
  "HttpMethod": "POST",
  "Body": null,
  "BinaryContent": "BQ==",
  "ContentType": "application/octet-stream",
  "ExpectedHtppCode": 200,
  "TokenHeaderName": "Ocp-Apim-Subscription-Key",
  "Cost": {
    "ServiceName": "FaceApi",
    "Cost": 1.0,
    "Unit": "tr"
  },
  "CognitiveServiceDoc": "https://westus.dev.cognitive.microsoft.com/docs/services/563879b61984550e40cbbe8d/operations/563879b61984550f30395236"
}
